docs/tutorials/start-a-private-network-with-substrate.md

```diff
@@ -145,77 +118,107 @@ yarn
+```
 
-Start Your Private Blockchain
+Creating the chainspec
 -----------------------------
 Last time around, we used `--chain=local` which is a predefined "chainspec" that has Alice and Bob specified as validators along with many other useful defaults.
 
-### Creating the chainspec
 Rather than writing our chainspec completely from scratch, we'll just make a few modifications to the the one we used before. To start we need to export the chainspec to a json file. Remember, further details about all of these commands are available by running `node-template --help`.
 
 ```bash
 ./target/release/node-template build-spec --chain=local > customSpec.json
 ```
 
-The file we just created contains several fields, and one can learn a lot by exploring them. By far the largest field is a single hex number that is the Wasm binary of our runtime. It is what you built earlier when you ran the `build.sh` script. Learn more about why it's useful to have that info on-chain from [Gav's web3 summit demo](https://www.youtube.com/watch?v=0IoUZdDi5Is).
+The file we just created contains several fields, and one can learn a lot by exploring them. By far the largest field is a single hex number that is the Wasm binary of our runtime. It is part of what you built earlier when you ran the `cargo build` command. Having this inforamtion on-chain allows supports forkless upgrades.
 
-The portion of the file we're interested in is the authorities which looks like this
+The portion of the file we're interested in is the babe authorities (used for creating blocks) and the grandpa authorieits (used for finalizing blocks). That section looks like this
 ```json
-  "consensus": {
-      "authorities": [
-        "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
-        "5GoNkf6WdbxCFnPdAnYYQyCjAKPJgLNxXwPjwTh6DGg6gN3E",
-      ],
-      "code": <snip>
-    }
+"babe": {
+  "authorities": [
+    [
+      "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
+      1
+    ],
+    [
+      "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
+      1
+    ]
+  ]
+},
+"grandpa": {
+  "authorities": [
+    [
+      "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
+      1
+    ],
+    [
+      "5GoNkf6WdbxCFnPdAnYYQyCjAKPJgLNxXwPjwTh6DGg6gN3E",
+      1
+    ]
+  ]
+},
+
 ```
 
-All we need to do is change the authority addresses listed (currently Alice and Bob) to our own addresses that we generated in the previous step. A single person should do these steps and share the resulting file. with their fellow validators. (Because Rust -> Wasm builds aren't "reproducible", each person will get a slightly different Wasm blob which will break consensus if each participant generates the file themselves.)
+All we need to do is change the authority addresses listed (currently Alice and Bob) to our own addresses that we generated in the previous step. The sr25519 addresses go in the `babe` section, and the ed25519 addresses in the `grandpa` section. You may add as many validators as you like.
```

Comment:
This is a wiki in polkadot protocol has some detail: https://wiki.polkadot.network/en/latest/polkadot/learn/keys/
Not sure if we want to link it here.

Reply:
I think it would be good to link it. Something like, "If you're interested in reading more about the key pairs, go to ...".